android/src/main/java/com/reactnativecommunity/webview/RNCWebViewClient.java

```diff
@@ -81,6 +81,8 @@ public void onPageStarted(WebView webView, String url, Bitmap favicon) {
       mLastLoadFailed = false;
 
       RNCWebView reactWebView = (RNCWebView) webView;
+      
+      reactWebView.getSettings().setUserAgentString(reactWebView.getUserAgentString());
       reactWebView.callInjectedJavaScriptBeforeContentLoaded();
```

Comment:
Considering that `injectedJavaScriptBeforeContentLoaded` does not work reliably on Android, I'm a bit nervous that this would make setting the user agent not work reliably anymore either...

Reply:
Isn't the reason why `injectedJavaScriptBeforeContentLoaded` didn't work reliably because of `evaluationJavaScript()`?

Is `onPageStarted` also relevant to this issue?

Reply:
I don't think the reason has ever been fully understood, but there was at one point a theory that it was related to the lifecycle events: https://github.com/react-native-webview/react-native-webview/pull/1099#issuecomment-645678919

Reply:
After some research, we come up with a different solution.

https://github.com/react-native-webview/react-native-webview/pull/3133